src/sql/azdata.proposed.d.ts

```diff
@@ -82,11 +82,11 @@ declare module 'azdata' {
 			/**
 			 * URI of the notebook document that is sending this execute request.
 			 */
-			notebookUri: vscode.Uri;
+			notebookUri: string;
```

Comment:
I was having problems earlier because calling toString() on a URI wouldn't return something like "untitled:Notebook-0". Instead it would return the default "Object" string for when no toSring() behavior is defined. When I inspected the URI fields, I noticed there were also slightly different fields than a normal URI, and this change in structure seemed to happen after passing the URI over an RPC proxy. It wouldn't actually work as a URI until you called URI.revive() on it, despite being typed as a vscode URI.

Reply:
I'm guessing that's why all our other "URI" fields in the azdata APIs are actually just strings. Turning that back into a URI at the eventual destination worked a lot better.

Reply:
Yes, so when it passes through the RPC channel it loses stuff like functions. That's why you see the `UriComponents`.

You need to always call `Uri.revive` on the other side before attempting to use any of the class functionality. 

Did you try fixing that first? We shouldn't need to be converting to strings at any point during the transition.

Reply:
I can revert that change, but currently there are 104 occurrences of string URIs in azdata.d.ts, and another 4 more in proposed. It's seems to have become an established pattern. It's used all over the place in the Query code, for example.

Reply:
It's not a good pattern and we shouldn't be doing it unless it's actually necessary.